libbeat/processors/add_cloud_metadata/provider_aws_ec2.go

```diff
@@ -94,8 +95,20 @@ func fetchRawProviderMetadata(
 ) {
 	logger := logp.NewLogger("add_cloud_metadata")
 
+	// generate AWS specific client with overriding requirements
+	var awsHTTPClient awshttp.BuildableClient
+	awsHTTPClient = *awsHTTPClient.WithTimeout(client.Timeout)
+	awsHTTPClient = *awsHTTPClient.WithTransportOptions(func(tr *http.Transport) {
```

Comment:
Do we need both `awsHTTPClient.WithTimeout` and `awsHTTPClient.WithTransportOptions` here?

Reply:
Good point, I did this here to match the derived client to the client we parsed into this function. And this generic client is derived here [1] and have some overriding options :) 

[1] - https://github.com/elastic/beats/blob/main/libbeat/processors/add_cloud_metadata/providers.go#L164-L175

Reply:
Maybe something like this then? 
```
	awsHTTPClient := *awshttp.NewBuildableClient().WithTimeout(client.Timeout).WithTransportOptions(func(tr *http.Transport) {
		transport, ok := client.Transport.(*http.Transport)
		if ok {
			tr.TLSClientConfig = transport.TLSClientConfig
		}

		tr.DisableKeepAlives = true
	})
```

Reply:
Yeah that's a nice formatting. I have adopted it :)